lib/common/strings.c

```diff
@@ -148,6 +150,129 @@ crm_parse_int(const char *text, const char *default_text)
     return (int) result;
 }
 
+/*!
+ * \internal
+ * \brief Scan a double-precision floating-point value from a string
+ *
+ * \param[in]  text      String to scan
+ * \param[out] result    If not NULL, where to store scanned value
+ * \param[out] end_text  If not NULL, where to store pointer to just
+ *                       after value
+ *
+ * \return Standard Pacemaker return code (also set errno on error)
```

Comment:
We could make the doc comment more helpful by specifying the values of errno that might be set -- namely, EINVAL for bad input and ERANGE for overflow/underflow. We can do the same for scan_ll(), crm_parse_ll(), crm_parse_int(), and pcmk_parse_double().

Would those details go in the \return section?

Reply:
Come to think of it, returning anything at all seems redundant, since rc is always equal to errno. It seems that one or the other would be fine. I used both because scan_ll() uses both.

Reply:
scan_ll() sets errno for backward API compatibility, but going forward for new functions we only need to use the standard return code.

Specific values can be documented in either \return or a \note. See pcmk__remote_ready() for a trivial example. A series of \retval entries is also fine if all values are enumerated, but since we're still transitioning between the legacy and standard rc's I prefer \return so we can say which it is.